### ISSUE TITLE
[c] tint black color error

### ISSUE THREAD

**Issue by huangwei1024** [EsotericSoftware/spine-runtimes#1097]
#1090 
sorry for error tag, I test the problem in cocos2dx, it use spine-c runtime.

![image](https://user-images.githubusercontent.com/667876/38659278-794037f4-3e5b-11e8-850c-001c4efacf37.png)

json on the left, binary on the right

```json
{ "name": "shenti1", "bone": "shenti1", "color": "ff0000ff", "dark": "0000ff", "attachment": "shenti1" },
```
blue dark color define in json. but not correct in binary file.

**Comment by huangwei1024** [comment]
```json
"skeleton": { "hash": "lFI7Eo/wTi1X1631kLdFsalqCTE", "spine": "3.6.52", "width": 154.81, "height": 175.5 },
```

![image](https://user-images.githubusercontent.com/667876/38659782-8147e954-3e5d-11e8-9da0-08380d7c77cb.png)


the export file version is 3.6.52, spine-c runtime is branch/3.6 e25989ca6d88a15876c360ac2949b5469c4d6054

**Comment by huangwei1024** [comment]
I modified in my project. https://github.com/EsotericSoftware/spine-runtimes/issues/1090#issue-307724445

![image](https://user-images.githubusercontent.com/667876/38662127-dc022276-3e65-11e8-98c9-efef2e8f0983.png)

its look ok. but I dont known the problem at the binary file or the code `spSkeletonBinary_readSkeletonDataFile`

**Comment by badlogic** [comment]
Could you try the 3.7-beta branch? I've added fixes for the parsing there https://github.com/EsotericSoftware/spine-runtimes/commit/b40eb593c978d6ee47d4c67bf23ee78714e0c03a.

**Comment by huangwei1024** [comment]
you test yout code ok?

![image](https://user-images.githubusercontent.com/667876/38664260-a507ad08-3e6b-11e8-999b-2184a6e2b55d.png)
![image](https://user-images.githubusercontent.com/667876/38664270-aa3cbb88-3e6b-11e8-82fc-3c13c312712d.png)

the rgb bytes sequence not same as you, but I test my code is ok.

**Comment by badlogic** [comment]
Terribly sorry, you are of course correct. I've fixed this in the 3.7-beta branch and the 3.6 branch!
